### PR TITLE
Move Version property to config.props & output to buildinfo.json

### PR DESCRIPTION
**Pull request by donnie-msft**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/2167

Regression? Last working version:

## Description

After moving this, my understanding is that `$(Version)` will contain the full package # that we insert into VS, so that it can be used in our insertion PRs.
I have already modified the NuGet Client Release to modify the following files, but I need the full version# of VS to put here:
- `nuget.packageconfig`
- `NuGetVisualStudio.props`

I don't know how to test this until I have an Official build to try the release on.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: build/config.props [hunk 1]
```diff
@@ -37,6 +37,35 @@
     <VsTargetChannel Condition="'$(IsEscrowMode)' == 'true'">int.d$(VsTargetMajorVersion).$(MinorNuGetVersion)</VsTargetChannel>
   </PropertyGroup>
 
+ <PropertyGroup Condition=" '$(PreReleaseVersion)' == '' ">
```
Comment (nkolev92) [inline build/config.props:40]: This is just step 1. 

You basically need to get the exact version to buildinfo.json, which is generated here: https://github.com/NuGet/NuGet.Client/blob/0ebb20188285663d7c8349df9ddfd71020c231c4/scripts/cibuild/ConfigureVstsBuild.ps1#L170-L189.

Comment (nkolev92) [reply build/config.props:40]: You need an equivalent target as the GetNuGetSdkVsSemanticVersion and the likes

Comment (donnie-msft) [reply build/config.props:40]: If I can get the version there, why do I need this change?

Comment (nkolev92) [reply build/config.props:40]: The targets and props are defined in config.props for all other properties. common.project.props has other imports and properties set that you don't want or need.

Comment (donnie-msft) [reply build/config.props:40]: How's this?
image

Comment (nkolev92) [reply build/config.props:40]: I think that's what you want.

**Review by nkolev92 — APPROVED** [review]

**Pull request merged**